src/v/model/record.h

```diff
@@ -532,11 +532,13 @@ using tx_seq = named_type<int64_t, struct tm_tx_seq>;
 using producer_id = named_type<int64_t, struct producer_identity_id>;
 using producer_epoch = named_type<int16_t, struct producer_identity_epoch>;
 
+static constexpr producer_epoch no_producer_epoch{-1};
+static constexpr producer_id no_producer_id{-1};
 struct producer_identity
   : serde::
       envelope<producer_identity, serde::version<0>, serde::compat_version<0>> {
-    int64_t id{-1};
-    int16_t epoch{0};
+    producer_id id{no_producer_id};
+    producer_epoch epoch{no_producer_epoch};
```

Comment:
Default was 0 and now it is -1, is this equivalent? I think producer_identity without epoch is for idempotent producers, won't there be compatibility problems with old snapshots etc.?

Reply:
yes, the default was incorrect, i have checked the use of this type and changing the value here isn't a problem